sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -145,6 +150,11 @@ async def send(self, request: HttpRequest, **config: Any) -> Optional[AsyncHttpR
         :keyword str proxy: will define the proxy to use all the time
         """
         await self.open()
+        try:
+            auto_decompress = self.session.auto_decompress
```

Comment:
`self.session` is supposed to be an `aiohttp.ClientSession`, correct? If so, this will never raise an `AttributeError`.

Reply:
Btw, not sure why we are missing type annotations for `self.session`. I would expect us to have that!

Reply:
This is to handle custom session. I suppose it is possible for a custom session that does not have attribute auto_decompress and I will regard custom session as auto_decompress enabled.